tests/nnstreamer_filter_flare/unittest_filter_flare.cc

```diff
@@ -0,0 +1,348 @@
+/**
+ * @file	unittest_filter_flare.cc
+ * @date	23 May 2025
+ * @brief	Unit test for tensor filter flare plugin
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Hyunil Park <hyunil46.park@samsung.com>
+ * @bug		No known bugs.
+ */
+
+#include <gtest/gtest.h>
+#include <gst/app/gstappsink.h>
+#include <gst/app/gstappsrc.h>
+#include <gst/gst.h>
+#include <unittest_util.h>
+
+/* Static variable to track the number of async callback invocations. */
+static int new_sample_count = 0;
+
+/**
+ * @brief Callback function for appsink element to pull sample and print the output text from llamacpp plugin
+ */
+static GstFlowReturn
+new_sample_cb (GstElement *sink, gpointer user_data)
+{
+  GstSample *sample;
+  GstBuffer *buffer;
+  GstMapInfo map;
+
+  new_sample_count++;
+  g_signal_emit_by_name (sink, "pull-sample", &sample);
+  if (sample) {
+    buffer = gst_sample_get_buffer (sample);
+    if (gst_buffer_map (buffer, &map, GST_MAP_READ)) {
+      g_printf ("%.*s", (int) map.size, (char *) map.data);
+      gst_buffer_unmap (buffer, &map);
+    }
+    gst_sample_unref (sample);
+    return GST_FLOW_OK;
+  }
+  return GST_FLOW_ERROR;
+}
+
+/**
+ * @brief Test fixture class for tensor_filter Flare plugin unit tests.
+ */
+class NNStreamerFilterFlareCppTest : public ::testing::Test
+{
+  protected:
+  GstElement *pipeline, *appsrc, *appsink, *tensor_filter;
+  gboolean loaded;
+  gboolean skip_test;
+
+  gchar *model;
+  gchar *data_file;
+  gchar *pipeline_str;
+  gchar *tokenizer;
+
+  public:
+  /**
+   * @brief Constructor a new NNStreamerFilterFlareCppTest object
+   */
+  NNStreamerFilterFlareCppTest ()
+  {
+    loaded = FALSE;
+    skip_test = FALSE;
```

Comment:
Initialize model, data_file, pipeline_str, tokenizer as NULL.
Without such, this may incur segfault (when skip_test becomes TRUE)